guacamole-common-js/src/main/webapp/modules/SessionRecording.js

```diff
@@ -684,14 +766,13 @@ Guacamole.SessionRecording = function SessionRecording(source) {
             // Pull the upcoming frame
             var next = frames[currentFrame + 1];
 
-            // Calculate the real timestamp corresponding to when the next
-            // frame begins
-            var nextRealTimestamp = next.timestamp - startVideoTimestamp + startRealTimestamp;
+            // Add the offset to the next frame timestamp to calculate the delay
+            var delay = toRelativeTimestamp(next.timestamp) - currentPosition;
```

Comment:
So to be clear - you're suggesting that we modify `seekToFrame` to either advance to the next frame _or_ increment `currentPosition` depending on whether `currentPosition + refreshInterval` >= the timestamp of the next frame? 

I think that could make sense. Let me see if it's simpler than what I have now.

Reply:
Or rather, not the `seekToFrame` function iself, but perhaps a wrapper function here.

Reply:
> So to be clear - you're suggesting that we modify `seekToFrame` to either advance to the next frame _or_ increment `currentPosition` depending on whether `currentPosition + refreshInterval` >= the timestamp of the next frame?

More or less. My thinking was:

* The core issue is that the previous code scheduled the next seek based on the frame duration, which can be significantly longer than one second.
* Part of these changes is ensuring we can reliably seek to any position, even between frames.
* We might be able to achieve the desired effect by ensuring the delay until next seek is bounded at the refresh interval, even if that seek would be mid-frame.

I'd think that would mean we'd have to:

1. Calculate the timestamp of the nearest whole multiple-of-`refreshInterval`.
2. If the next frame is before that, easy - seek to _that_.
3. If not, seek to the calculated timestamp.

That way we are guaranteed to be updated every `refreshInterval` relative to playback position, not relative to a local timer.

Reply:
How do my latest changes look? That's more or less what I'm doing, though I didn't do do the multiple-of-`refreshInterval` thing, I just seek to the next `refreshInterval`, or the next frame, whichever is sooner.

Reply:
Ok, resolved.

Reply:
> How do my latest changes look? That's more or less what I'm doing, though I didn't do do the multiple-of-`refreshInterval` thing, I just seek to the next `refreshInterval`, or the next frame, whichever is sooner.

I think you might need to round down to the nearest multiple-of-`refreshInterval`. Seeking to the next `refreshInterval` would be relative to the current position, which may not be correct. For example (timestamps below are relative to start of recording):

1. Frame at 0ms. Player shows "00:00".
2. Frame at 400ms. Player shows "00:00".
3. Frame at 10000ms. Player waits until current position + `refreshInterval`, showing "00:01" at 1400ms.

Here, the player took 1.4 seconds of playback before the visible timestamp progressed from "00:00" to "00:01". If this sort of issue occurred multiple times in the recording, it would appear to the user as a jittering playback timestamp, incrementing at inconsistent intervals.

With rounding down to the nearest multiple of `refreshInterval`, we instead get:

1. Frame at 0ms. Player shows "00:00".
2. Frame at 400ms. Player shows "00:00".
3. Frame at 10000ms. Player waits until the nearest multiple of `refreshInterval`, showing "00:01" at 1000ms.

Reply:
Ok, how does this look?